### PR TITLE
Describe image population

### LINKED ISSUE #41: Document image population
# Summary

Document in the architecture section how the image population process is working.

/refs
https://kubernetes.io/blog/2021/08/30/volume-populators-redesigned/
https://github.com/kubernetes-csi/lib-volume-populator

### PR DESCRIPTION
**Pull request by afritzler**
# Proposed Changes

- Main readme cleanup
- Describe the image population process

Fixes #41 

/cc @EvgeniaOrfeeva

### CODE REVIEW

File: docs/architecture/README.md [hunk 1]
```diff
@@ -1,15 +1,51 @@
...
-    as a tenant separation. All `Volumes` within a `Namespace` belong essentially to one tenant.
+    As Ceph does not support the creation of access secrets for each individual block devices, we chose to use 
+    Kubernetes `Namespace`s as a tenant separation. All `Volumes` within a `Namespace` belong essentially to one tenant.
 
-The task of the `cephlet` is to create a Ceph block device for every `Volume` in a given namespace. Additionally,
-for every `Namespace` an own `CephClient` and a corresponding `StorageClass` is created. The `StorageClass` is being used
-to later create the `PersistentVolumeClaims` for a given `Namespace`. The access credentials which are being extracted
-from the `CephClient` are stored in a `Secret` which is then referenced in the status of the `Volume`.
+Additionally, for every `Namespace` an own `CephClient` and a corresponding `StorageClass` is created. The `StorageClass` 
+is being used to later create the `PersistentVolumeClaims` for a given `Namespace`. The access credentials which are being
```
Comment (lukasfrank) [inline docs/architecture/README.md:47]: for a given `Namespace` => in a given ceph namespace.

File: docs/architecture/README.md [hunk 2]
```diff
@@ -1,15 +1,51 @@
...
+    As Ceph does not support the creation of access secrets for each individual block devices, we chose to use 
+    Kubernetes `Namespace`s as a tenant separation. All `Volumes` within a `Namespace` belong essentially to one tenant.
 
-The task of the `cephlet` is to create a Ceph block device for every `Volume` in a given namespace. Additionally,
-for every `Namespace` an own `CephClient` and a corresponding `StorageClass` is created. The `StorageClass` is being used
-to later create the `PersistentVolumeClaims` for a given `Namespace`. The access credentials which are being extracted
-from the `CephClient` are stored in a `Secret` which is then referenced in the status of the `Volume`.
+Additionally, for every `Namespace` an own `CephClient` and a corresponding `StorageClass` is created. The `StorageClass` 
+is being used to later create the `PersistentVolumeClaims` for a given `Namespace`. The access credentials which are being 
+extracted from the `CephClient` are stored in a `Secret` which is then referenced in the status of the `Volume`.
```
Comment (lukasfrank) [inline docs/architecture/README.md:48]: to simplify the cleanup we create for each volume a secret and copy the content of the ceph client secret. 
additionally: the ceph client secret is in the "ceph default ns"

**Review by lukasfrank — CHANGES_REQUESTED** [review]

File: docs/usage/README.md [hunk 3]
```diff
@@ -74,16 +80,19 @@ status:
...
 NAME            TYPE     DATA   AGE
 sample-volume   Opaque   2      93s
 ```
 
 ## Rook resources
+
 Administrators can also observe the rook related resources. 
 Every (customer) namespace contain a `cephblockpoolradosnamespaces` and a `cephclients` resource. Under the hood rook 
 generates a rados namespace and granting access to it for the specific ceph client user.
+
```
Comment (EvgeniaOrfeeva) [inline docs/usage/README.md:95]:
```suggestion

Administrators can also observe the rook related resources. Every (customer) namespace contain a `cephblockpoolradosnamespaces` and a `cephclients` resource. Under the hood, rook generates a rados namespace and granting access to it for the specific ceph client user.

```

File: docs/architecture/README.md (hunk 2 above)
Comment (afritzler) [reply docs/architecture/README.md]: You are right I simplified this part here a bit.

**Comment by afritzler** [comment]
Thanks a lot @EvgeniaOrfeeva and @lukasfrank for your feedback.  Regarding the secret handling: I think we can go with the simplified version for now.

**Pull request merged**